packages/typespec-client-generator-core/test/packages/paged-operation.test.ts

```diff
@@ -15,33 +15,112 @@ describe("typespec-client-generator-core: paged operation", () => {
-            @Azure.Core.nextLink
-            @clientName("nextLink")
-            next: string;
-        }
-        model Test {
-            id: string;
-        }
-      `);
+  it("azure paged result with encoded name", async () => {
+    await runner.compileWithBuiltInService(`
+      op test(): ListTestResult;
+      @pagedResult
+      model ListTestResult {
+        @items
+        @clientName("values")
+        tests: Test[];
+        @Azure.Core.nextLink
+        @clientName("nextLink")
+        next: string;
+      }
+      model Test {
+        id: string;
+      }
+    `);
     const sdkPackage = runner.context.sdkPackage;
     const method = getServiceMethodOfClient(sdkPackage);
     strictEqual(method.name, "test");
     strictEqual(method.kind, "paging");
-    strictEqual(method.crossLanguageDefintionId, "My.Service.test");
     strictEqual(method.nextLinkPath, "nextLink");
 
     const response = method.response;
     strictEqual(response.kind, "method");
     strictEqual(response.resultPath, "values");
   });
+
+  it("normal paged result", async () => {
+    await runner.compileWithBuiltInService(`
+      @list
+      op test(): ListTestResult;
+      model ListTestResult {
+        @pageItems
+        tests: Test[];
+        @TypeSpec.nextLink
+        next: string;
+      }
+      model Test {
+        id: string;
+      }
+    `);
+    const sdkPackage = runner.context.sdkPackage;
+    const method = getServiceMethodOfClient(sdkPackage);
+    strictEqual(method.name, "test");
+    strictEqual(method.kind, "paging");
+    strictEqual(method.nextLinkPath, "next");
+
+    const response = method.response;
+    strictEqual(response.kind, "method");
+    strictEqual(response.resultPath, "tests");
+  });
+
+  it("normal paged result with encoded name", async () => {
+    await runner.compileWithBuiltInService(`
+      @list
+      op test(): ListTestResult;
+      model ListTestResult {
+        @pageItems
+        @clientName("values")
+        tests: Test[];
+        @TypeSpec.nextLink
+        @clientName("nextLink")
+        next: string;
+      }
+      model Test {
+        id: string;
+      }
+    `);
+    const sdkPackage = runner.context.sdkPackage;
+    const method = getServiceMethodOfClient(sdkPackage);
+    strictEqual(method.name, "test");
+    strictEqual(method.kind, "paging");
+    strictEqual(method.nextLinkPath, "nextLink");
+
+    const response = method.response;
+    strictEqual(response.kind, "method");
+    strictEqual(response.resultPath, "values");
+  });
+
+  // skip for current paging implementation does not support nested paging value
+  it.skip("normal paged result with nested paging value", async () => {
+    await runner.compileWithBuiltInService(`
+      @list
+      op test(): ListTestResult;
+      model ListTestResult {
+        results: {
+          @pageItems
+          values: Test[];
+        };
+        pagination: {
+          @TypeSpec.nextLink
+          nextLink: string;
```

Comment:
A thing not related to this PR.

What `nextLinkPath` be, if this property be
```
`next.link`: string;
```

Reply:
good catch. i think export it as an array is more reasonable.

Reply:
https://github.com/Azure/typespec-azure/issues/1920

Reply:
My guess is the design want to use JSON path. Though I am not sure how JSON path express that. And I guess not many language has built support for JSON path anyway...